jxl/src/var_dct/dct.rs

```diff
@@ -138,39 +217,71 @@ macro_rules! define_dct_1d {
                 //
                 //    Inputs: first N/2 rows of data, second N/2 rows of data
                 //    Output: first N/2 rows of tmp_buffer
-                CoeffBundle::<$nhalf, COLUMNS>::add_reverse(
+                CoeffBundle::<$nhalf, COLUMNS>::add_reverse::<D>(
+                    d,
                     &data[0..$nhalf],
                     &data[$nhalf..$n],
                     &mut tmp_buffer[0..$nhalf],
+                    starting_column,
+                    num_columns,
                 );
 
                 // 2. First Recursive Call (do_dct)
                 //    first half
-                DCT1DImpl::<$nhalf>::do_dct::<D, COLUMNS>(d, &mut tmp_buffer[0..$nhalf]);
+                DCT1DImpl::<$nhalf>::do_dct::<D, COLUMNS>(
```

Comment:
Replacing this with a macro invocation that either does a direct call or uses d.call() depending on $nhalf would be good for compilation times.